python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -478,7 +476,15 @@ def updatedQuantumInputs(self, quantum: Quantum, taskDef: TaskDef) -> Quantum:
         for key, refsForDatasetType in quantum.inputs.items():
             newRefsForDatasetType = updatedInputs[key]
             for ref in refsForDatasetType:
-                if ref.id is None:
+
+                # Inputs may already be resolved even if they do not exist, but
+                # we have to re-resolve them because IDs are ignored on output.
+                # If they already exist in datastore do not re-resolve them
+                # as it may change dataset type.
```

Comment:
As of DM-37450 (merged today) `findDataset` will respect the storage class of a `DatasetType` that is given to it.

Reply:
Hmm, I think I want to keep the datastore existence check here, otherwise I'd have to provide `timespan` to `findDataset` method which I don't know where to get. I'll update the comment.